source/slang/core.meta.slang

```diff
@@ -1276,950 +1273,979 @@ static const struct TextureTypePrefixInfo
-                sb << baseName;
-                sb << baseShapeName;
-                if (isMultisample) sb << "MS";
-                if (isArray) sb << "Array";
-    //                        if (isShadow) sb << "Shadow";
-                sb << "\n{";
-
-                char const* samplerStateParam = prefixInfo.combined ? "" : "SamplerState s, ";
-
-                if( !isMultisample )
+
+                if (includeMipInfo)
                 {
-                    sb << "float CalculateLevelOfDetail(" << samplerStateParam;
-                    sb << "float" << kBaseTextureTypes[tt].coordCount << " location);\n";
+                    sb << ", ($" << aa++ << " = textureQueryLevels($0))";
+                }
+
+
+                sb << ")\")\n";
+            }
+
+            sb << "[__readNone]\n";
+            sb << "void GetDimensions(";
+            if(includeMipInfo)
+                sb << "uint mipLevel, ";
+
+            switch(baseShape)
+            {
+            case TextureFlavor::Shape::Shape1D:
+                sb << t << "width";
+                break;
+
+            case TextureFlavor::Shape::Shape2D:
+            case TextureFlavor::Shape::ShapeCube:
+                sb << t << "width,";
+                sb << t << "height";
+                break;
+
+            case TextureFlavor::Shape::Shape3D:
+                sb << t << "width,";
+                sb << t << "height,";
+                sb << t << "depth";
+                break;
+
+            default:
+                assert(!"unexpected");
+                break;
+            }
+
+            if(isArray)
+            {
+                sb << ", " << t << "elements";
+            }
+
+            if(isMultisample)
+            {
+                sb << ", " << t << "sampleCount";
+            }
+
+            if(includeMipInfo)
+                sb << ", " << t << "numberOfLevels";
+
+            sb << ");\n";
+        }
+
+        // `GetSamplePosition()`
+        if( isMultisample )
+        {
+            sb << "float2 GetSamplePosition(int s);\n";
+        }
 
-                    sb << "float CalculateLevelOfDetailUnclamped(" << samplerStateParam;
-                    sb << "float" << kBaseTextureTypes[tt].coordCount << " location);\n";
+        // `Load()`
+
+        if( base.coordCount + isArray < 4 )
+        {
+            // The `Load()` operation on an ordinary `Texture2D` takes
+            // an `int3` for the location, where `.xy` holds the texel
+            // coordinates, and `.z` holds the mip level to use.
+            //
+            // The third coordinate for mip level is absent in
+            // `Texure2DMS.Load()` and `RWTexture2D.Load`. This pattern
+            // is repreated for all the other texture shapes.
+            //
+            bool needsMipLevel = !isMultisample && (access == SLANG_RESOURCE_ACCESS_READ);
+
+            int loadCoordCount = base.coordCount + isArray + (needsMipLevel?1:0);
```

Comment:
Uses bool as an int, with isArray, and uses `:?` (which was probly in original code). I actually think just using int(isArray) + int(needsMipLevel) is ok.